### PR TITLE
Feature: Added search via ?q= param

### PR DESCRIPTION
**Pull request by andrebrito16**
# Feature: Added Search via ?q= Param

## Overview
This pull request introduces a new feature allowing users to search directly via the URL using the `?q=` parameter. This enhances user experience by providing a more convenient way to initiate searches.

## Changes
- **Key Changes:**
    - Added `?q=` parameter functionality to the `/` route.
    - Implemented search functionality using the `useSearchParams` hook in `app/chatpage.tsx`.
    - Modified `app/actions.ts` to remove unnecessary logging and simplify the search logic.
- **New Features:**
    - Added a search form to `app/chatpage.tsx` for users to input search queries.
- **Refactoring:**
    - Removed unnecessary logging statements from `app/actions.ts`.
    - Improved code readability and organization in `app/chatpage.tsx`.

> ✨ Generated with love by [Kaizen](https://cloudcode.ai) ❤️

<details>
<summary>Original Description</summary>
Now users can search via /q=<search>
</details>

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,6 @@
+SEARCH_API_KEY=
+MEM0_API_KEY=
+BACKEND_SECURITY_KEY=
+GOOGLE_CLIENT_ID=
+GOOGLE_CLIENT_SECRET=
+OPENAI_API_KEY=
```
[the remark this answers is not in the document]
Comment (andrebrito16) [reply .env.example:6]: !feedback This is just a example of env file, does not contains any secret.

[the remark this answers is not in the document]
Comment (andrebrito16) [reply .env.example:6]: !review This is just a example of env file, does not contains any secret.

File: app/actions.ts [hunk 2]
```diff
@@ -10,29 +10,6 @@ export const getSearchResultsFromMemory = async (
 ): Promise<BingResults | null> => {
   if (!query || !user?.user) return null;
 
-  console.log("User's query: ", query, user);
-
-  const mem0Response = fetch('https://api.mem0.ai/v1/memories/', {
```
Comment (Dhravya) [inline app/actions.ts:15]: why did you remove this? this is imp for storing user informatino.

Comment (andrebrito16) [reply app/actions.ts:15]: Sorry, I do not see that. Will add again, one sec!

File: app/actions.ts [hunk 3]
```diff
@@ -10,9 +10,7 @@ export const getSearchResultsFromMemory = async (
...
+  await fetch('https://api.mem0.ai/v1/memories/', {
```
[the remark this answers is not in the document]
Comment (Dhravya) [reply app/actions.ts:13]: good bot. @andrebrito16 can you do this?

**Review by Dhravya — CHANGES_REQUESTED** [review]

**Review by Dhravya — APPROVED** [review]

**Comment by Dhravya** [comment]
LGTM thanks!
